src/dwarf.rs

```diff
@@ -952,11 +1005,114 @@ impl DwarfResolver {
     ///
     /// This function is pretty much the same as `find_line_as_ref()`
     /// except returning a copies of `String` instead of `&str`.
+    #[allow(dead_code)]
     pub fn find_line(&self, address: u64) -> Option<(String, String, usize)> {
         let (dir, file, line_no) = self.find_line_as_ref(address)?;
         Some((String::from(dir), String::from(file), line_no))
     }
 
+    /// Extract the symbol information from DWARf if having not did it
+    /// before.
+    fn ensure_debug_info_syms(&self) -> Result<(), Error> {
+        if self.enable_debug_info_syms {
+            let mut dis_ref = self.debug_info_syms.borrow_mut();
+            if dis_ref.is_some() {
+                return Ok(());
+            }
+            let mut debug_info_syms = debug_info_parse_symbols(&self.parser, None, 1)?;
+            debug_info_syms.sort_by_key(|v: &DWSymInfo| -> &str { v.name });
+            *dis_ref = Some(unsafe { mem::transmute(debug_info_syms) });
+        }
+        Ok(())
+    }
+
+    /// Find the address of a symbol from DWARF.
+    ///
+    /// # Arguments
+    ///
+    /// * `name` - is the symbol name to find.
+    /// * `ctx` - is the context giving additional parameters.
+    pub fn find_address(&self, name: &str, ctx: &FindAddrContext) -> Result<SymbolInfo, Error> {
```

Comment:
name is not unique for static functions, should Vec<SymbolInfo> be returned instead? Basically, from my POV the only difference between find_address and find_addrs_regex should be that the former accepts fixed name and the latter accepts regex, but everything else is exactly the same.

Also note discrepancy in naming address (not addr) vs addrs (not addresses). Let keep the naming consistent.

Reply:
I also want to change this part for a while.  I think it is time now.